internal/server/evaluation/evaluation.go

```diff
@@ -46,18 +70,97 @@ func (e *EvaluateServer) Variant(ctx context.Context, v *rpcEvaluation.Evaluatio
+	s.logger.Debug("variant", zap.Stringer("response", resp))
 	return ver, nil
 }
+
+// Boolean evaluates a request for a boolean flag and entity.
+func (s *Server) Boolean(ctx context.Context, r *rpcEvaluation.EvaluationRequest) (*rpcEvaluation.BooleanEvaluationResponse, error) {
+	resp := &rpcEvaluation.BooleanEvaluationResponse{}
+
+	flag, err := s.store.GetFlag(ctx, r.NamespaceKey, r.FlagKey)
+	if err != nil {
+		var errnf errs.ErrNotFound
+
+		if errors.As(err, &errnf) {
+			resp.Reason = rpcEvaluation.EvaluationReason_FLAG_NOT_FOUND_EVALUATION_REASON
+			return resp, err
+		}
+
+		resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return resp, err
+	}
+
+	if flag.Type != flipt.FlagType_BOOLEAN_FLAG_TYPE {
+		resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return resp, errs.ErrInvalidf("flag type %s invalid", flipt.FlagType_name[int32(flag.Type)])
+	}
+
+	rollouts, err := s.store.GetEvaluationRollouts(ctx, r.NamespaceKey, r.FlagKey)
+	if err != nil {
+		resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+		return resp, err
+	}
+
+	var lastRank int32
+
+	for _, rollout := range rollouts {
+		if rollout.Rank < lastRank {
+			resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+			return resp, errs.ErrInvalidf("rollout rank: %d detected out of order", rollout.Rank)
+		}
+
+		lastRank = rollout.Rank
+
+		if rollout.Percentage != nil {
+			// consistent hashing based on the entity id and flag key.
+			hash := crc32.ChecksumIEEE([]byte(r.EntityId + r.FlagKey))
+
+			normalizedValue := float32(int(hash) % 100)
+
+			// if this case does not hold, fall through to the next rollout.
+			if normalizedValue < rollout.Percentage.Percentage {
+				resp.Value = rollout.Percentage.Value
+				resp.Reason = rpcEvaluation.EvaluationReason_MATCH_EVALUATION_REASON
+				s.logger.Debug("percentage based matched", zap.Int("rank", int(rollout.Rank)), zap.String("rollout_type", "percentage"))
+
+				return resp, nil
+			}
+		} else if rollout.Segment != nil {
+			matched, err := s.evaluator.matchConstraints(r.Context, rollout.Segment.Constraints, rollout.Segment.SegmentMatchType)
+			if err != nil {
+				resp.Reason = rpcEvaluation.EvaluationReason_ERROR_EVALUATION_REASON
+				return resp, err
+			}
+
+			// if we don't match the segment, fall through to the next rollout.
+			if !matched {
+				continue
+			}
+
+			resp.Value = rollout.Segment.Value
+			resp.Reason = rpcEvaluation.EvaluationReason_MATCH_EVALUATION_REASON
+
+			s.logger.Debug("segment based matched", zap.Int("rank", int(rollout.Rank)), zap.String("segment", rollout.Segment.SegmentKey))
+
+			return resp, nil
+		}
+	}
+
+	// If we have exhausted all rollouts and we still don't have a match, return the default value.
+	resp.Reason = rpcEvaluation.EvaluationReason_DEFAULT_EVALUATION_REASON
+	resp.Value = flag.Enabled
+	s.logger.Debug("default rollout matched", zap.Bool("value", flag.Enabled))
+
+	return resp, nil
+}
```

Comment:
We should be encapsulating the boolean evaluation logic in a boolean flag evaluator. Like how the multi-variant one is encapsulated. Then having the server just be dumn and invoke it after a bit of payload validation.